frontend/js/src/user/stats/components/UserArtistEvolutionActivity.tsx

```diff
@@ -0,0 +1,569 @@
+      "December",
+    ];
+  } else {
+    const nowYear = new Date().getFullYear();
+    const yearsWithData = rawData.reduce<number[]>((acc, item) => {
+      const y = parseInt(String(item.time_unit), 10);
+      if (!Number.isNaN(y) && (item.listen_count || 0) > 0) {
+        acc.push(y);
+      }
+      return acc;
+    }, []);
+    if (yearsWithData.length) {
+      const firstYearWithData = Math.min(...yearsWithData);
+      offsetYear = firstYearWithData - 1;
+      const years: string[] = [];
+      for (let y = offsetYear; y <= nowYear; y += 1) {
+        years.push(String(y));
+      }
+      orderedTimeUnits = years;
+    } else {
+      orderedTimeUnits = [];
+    }
+  }
+
+  const chartData: StreamDataItem[] = orderedTimeUnits.map((tu) => {
+    const timeData = groupedByTime[tu] || {};
+    const row: StreamDataItem = { id: tu };
+    topArtists.forEach((artist) => {
+      row[artist] = timeData[artist] ?? 0;
+    });
+    return row;
+  });
+
+  return {
+    chartData,
+    keys: topArtists,
+    orderedTimeUnits,
+    offsetYear,
+    artistMap,
+  };
+};
+
+export const artistEvolutionQueryKey = (
+  userName: string | undefined,
+  range: UserStatsAPIRange
+) => ["userArtistEvolutionActivity", userName, range] as const;
+
+export default function ArtistEvolutionActivityStreamGraph(
+  props: UserArtistEvolutionActivityProps
+) {
+  const { APIService } = React.useContext(GlobalAppContext);
+  const { user, range } = props;
+  const navigate = useNavigate();
+
+  const { data: loaderData, isLoading: loading } = useQuery({
+    queryKey: artistEvolutionQueryKey(user?.name, range),
+    queryFn: async () => {
+      try {
+        const queryData = (await APIService.getUserArtistEvolutionActivity(
+          user?.name,
+          range
+        )) as UserArtistEvolutionActivityResponse;
+        return { data: queryData, hasError: false, errorMessage: "" };
+      } catch (error) {
+        return {
+          data: {
+            payload: {
+              artist_evolution_activity: [],
+              offset_year: 2020,
+              range,
+              from_ts: 0,
+              to_ts: 0,
+              last_updated: 0,
+              user_id: user?.name ?? "",
+            },
+          } as UserArtistEvolutionActivityResponse,
+          hasError: true,
+          errorMessage: (error as Error).message,
+        };
+      }
+    },
+  });
+
+  const {
+    data: rawData = {
+      payload: {
+        artist_evolution_activity: [],
+        range,
+        from_ts: 0,
+        to_ts: 0,
+        last_updated: 0,
+        user_id: user?.name ?? "",
+      },
+    } as UserArtistEvolutionActivityResponse,
+    hasError = false,
+    errorMessage = "",
+  } = loaderData || {};
+
+  const isMobile = useMediaQuery("(max-width: 767px)");
+
+  const [topN, setTopN] = React.useState<number>(10);
+  const onTopNChange: React.ChangeEventHandler<HTMLSelectElement> = (e) => {
+    setTopN(parseInt(e.target.value, 10));
+  };
```

Comment:
This was supposedly introduced to implement the topN select, but the implementation was changed to use a dropdown instead. 
I assume it was changed to follow the other dropdowns on the page, but I think the select is about perfect for this job, and has better accessibility support. (we should probably change the other dropdown to a select, to be honest.)

And so concise after a small refactor!
23 lines reduced to 4:
```tsx
<select className="form-select" onChange={onTopNChange}>
  {[5, 10, 15].map((num) => {
    return <option value={num}>{num}</option>;
  })}
</select>
```

<img width="299" height="255" alt="Image" src="https://github.com/user-attachments/assets/8259f72e-03ae-42d0-b51e-747b233735c0" />

If you still want to use a dropdown instead, you can remove this unused function